From 14d28e7908d5421f15f9b94f4f37d66f14c4222e Mon Sep 17 00:00:00 2001
From: "Jasper St. Pierre" <jstpierre@mecheye.net>
Date: Wed, 22 Oct 2014 18:44:16 -0700
Subject: main: Don't update the PangoContext in clutter_set_font_flags

clutter_set_font_flags already calls clutter_backend_set_font_options,
which emits a signal which our PangoContext listens to, so this is just
duplicate and unneeded code.

https://bugzilla.gnome.org/show_bug.cgi?id=739050

diff --git a/clutter/clutter-main.c b/clutter/clutter-main.c
index 444ceba..3b9385a 100644
--- a/clutter/clutter-main.c
+++ b/clutter/clutter-main.c
@@ -3276,7 +3276,6 @@ clutter_clear_glyph_cache (void)
 void
 clutter_set_font_flags (ClutterFontFlags flags)
 {
-  ClutterMainContext *context = _clutter_context_get_default ();
   CoglPangoFontMap *font_map;
   ClutterFontFlags old_flags, changed_flags;
   const cairo_font_options_t *font_options;
@@ -3326,10 +3325,6 @@ clutter_set_font_flags (ClutterFontFlags flags)
   clutter_backend_set_font_options (backend, new_font_options);
 
   cairo_font_options_destroy (new_font_options);
-
-  /* update the default pango context, if any */
-  if (context->pango_context != NULL)
-    update_pango_context (backend, context->pango_context);
 }
 
 /**
-- 
cgit v0.10.1
 
